build/Makefile

```diff
@@ -443,42 +446,41 @@ setup-prometheus: PVC ?= true
 setup-prometheus: PV_SIZE ?= 64Gi
 setup-prometheus: SCRAPE_INTERVAL=30s
 setup-prometheus:
-	$(DOCKER_RUN) \
-		helm upgrade --install --wait prom stable/prometheus --namespace metrics \
+	$(DOCKER_RUN) helm repo add stable https://kubernetes-charts.storage.googleapis.com/
+	$(DOCKER_RUN) helm repo update
+	$(DOCKER_RUN) helm upgrade prom stable/prometheus \
+		--install --wait --namespace metrics --create-namespace \
 		--set server.global.scrape_interval=$(SCRAPE_INTERVAL),server.persistentVolume.enabled=$(PVC),server.persistentVolume.size=$(PV_SIZE) \
 		$(HELM_ARGS) \
 		-f $(mount_path)/build/prometheus.yaml
 
 uninstall-prometheus: $(ensure-build-image)
 	$(DOCKER_RUN) \
-		helm delete --purge prom
+		helm uninstall prom --namespace=metrics
 
 # setup grafana in the current cluster with datasource and dashboards ready for use with agones
 # by default Persistent Volume Claims are requested.
 setup-grafana: PVC ?= true
 setup-grafana: PV_SIZE ?= 64Gi
 setup-grafana: PASSWORD ?= admin
 setup-grafana:
+	-$(DOCKER_RUN) kubectl create namespace metrics
```

Comment:
Why does this one create the namespace here instead of passing the `--create-namespace` flag like the other  `helm upgrade` commands?

Reply:
It looks like it's because of the kubectl apply before the helm upgrade.... Do we have to do it in that order or could we do the apply after the helm upgrade to remove this line?

Reply:
Also, what does the `-` at the start of this line for?

Reply:
> It looks like it's because of the kubectl apply before the helm upgrade

I believe that is the reason yes. 

> Do we have to do it in that order or could we do the apply after the helm upgrade to remove this line?

I am honestly unsure (not that familiar with grafana), I'm just matching the order of what was done previously. I also have a strong feeling there was a long standing bug that if you didn't install prometheus first (which would create the namespace), then the namespace for grafana wouldn't exist, and this would fail.

> Also, what does the - at the start of this line for?

That means that if this command fails for any reason, the error will be ignored by the Makefile, and commands will continue. It's a nice way of saying "create this namespace if it doesn't already exist"

Reply:
> Presumably helm does that for us as well (in this new flow) so it would be awesome if we could make this look like the other lines. I'm not sure if it's worth trying to fix in this PR though, or maybe tinker with it as a follow-up?

Reply:
:shrug: maybe tinker with it in a subsequent PR?

Reply:
sgtm.